server/data/data-cassandra/src/main/java/org/apache/james/sieve/cassandra/CassandraSieveRepository.java

```diff
@@ -59,10 +60,10 @@ public CassandraSieveRepository(CassandraSieveDAO cassandraSieveDAO, CassandraSi
     }
 
     @Override
-    public DateTime getActivationDateForActiveScript(User user) throws ScriptNotFoundException {
-        return cassandraActiveScriptDAO.getActiveSctiptInfo(user).join()
+    public ZonedDateTime getActivationDateForActiveScript(User user) throws ScriptNotFoundException {
+        return ZonedDateTime.ofInstant(cassandraActiveScriptDAO.getActiveSctiptInfo(user).join()
             .orElseThrow(ScriptNotFoundException::new)
-            .getActivationDate();
+            .getActivationDate(), ZoneId.systemDefault());
```

Comment:
That's something dangerous, the behavior of the code depends on the timezone of the server. Is it what DateTime was doing before ? If yes, we may keep the code you wrote but open a bug about that.

Reply:
Well, indirectly, we stored a `new Date()`, which have the the system TimeZone and we fetch it assuming that it's the same TimeZone. So, it's a yet-to-come bug.

Reply:
I think my version makes the confusion more obvious.

Reply:
I was wrong, it's store in UTC, I'll handle it in the DAO.